40-car-detector/README.md

```diff
@@ -0,0 +1,229 @@
+
+If you're unhappy with the performance of the model, we have two suggestions:
+
+1. Collect more training data.
+2. Try out some of the "Expert mode tips" from [this Edge Impulse article](https://docs.edgeimpulse.com/docs/edge-impulse-studio/learning-blocks/object-detection/fomo-object-detection-for-constrained-devices#expert-mode-tips). Specifically, we found that increasing the `object_weight` parameter from 100 to 1000 was extremely helpful in getting better model performance.
+
+## Deploying the Model
+
+With your model trained, you'll now deploy it to the OpenMV cam's MCU.
+
+1. With your Edge Impulse project open, click Deployment in the left-hand toolbar.
+2. Search for "openmv" in the deployment options search bar and pick OpenMV Firmware.
+3. Click Build and wait for the download to complete.
+4. Follow [the "Deploying your impulse as an OpenMV firmware" section of this Edge Impulse article](https://docs.edgeimpulse.com/docs/deployment/running-your-impulse-openmv#deploying-your-impulse-as-an-openmv-firmware) to get the model onto your microcontroller using the OpenMV IDE.
+
+## Notehub Setup
+
+In order to push data from the Notecard to the cloud, you need to sign up for a free account on [notehub.io](https://notehub.io) and [create a new project](https://dev.blues.io/quickstart/notecard-quickstart/notecard-and-notecarrier-pi/#set-up-notehub).
+
+## MicroPython Code
+
+The script that will run on the MCU is `main.py`. It depends on [note-python](https://github.com/blues/note-python), a Python library for communicating with a Notecard.
+
+### note-python
+
+To get the note-python files onto the MCU, use the `setup_board.py` script. First, you must identify the MCU's serial port. On Linux, it'll typically be something like `/dev/ttyACM0`. Once you have that, run `python setup_board.py <serial port>`, replacing `<serial port>` with your serial port. This script does a few things:
+
+1. Clones note-python from GitHub.
+2. Creates the `/lib` and `/lib/notecard` directories on the MCU.
+3. Copies the `.py` files from `note-python/notecard` on your development machine to `/lib/notecard` on the MCU.
+4. Lists the contents of `/lib/notecard` so you can verify that everything was copied over.
+
+### `main.py`
+
+`main.py` loops infinitely, grabbing an image from the camera and running it through your model. The output of the model is a probability value in the range [0, 1], with 0 corresponding to a 0% probability of a car being in the image and 1 corresponding to a 100% probability. If this probability exceeds a configurable threshold (discussed further below), a car "event" begins. The event ends once the probability has dropped below the threshold for 12 consecutive images.*
+
+Once an event ends, a note is added to the [Notefile](https://dev.blues.io/api-reference/glossary/#notefile) `events.qo` in this format:
```

Comment:
Very, very trivial but “events.qo Notefile” sounds nicer than “Notefile events.qo” to me.